test/unit/api/datatype_api_black.h

```diff
@@ -57,6 +58,77 @@ void DatatypeBlack::testMkDatatypeSort()
   TS_ASSERT_THROWS_NOTHING(nilConstr.getConstructorTerm());
 }
 
+void DatatypeBlack::testMkDatatypeSorts()
+{
+  /* Create two mutual datatypes corresponding to this definition
+   * block:
+   *
+   *   DATATYPE
+   *     tree = node(left: tree, right: tree) | leaf(data: list),
+   *     list = cons(car: tree, cdr: list) | nil
+   *   END;
+   */
+  // Make unresolved types as placeholders
+  std::set<Sort> unresTypes;
+  Sort unresTree = d_solver.mkUninterpretedSort("tree");
+  Sort unresList = d_solver.mkUninterpretedSort("list");
+  unresTypes.insert(unresTree);
+  unresTypes.insert(unresList);
+
+  DatatypeDecl tree = d_solver.mkDatatypeDecl("tree");
+  DatatypeConstructorDecl node("node");
+  DatatypeSelectorDecl left("left", unresTree);
+  node.addSelector(left);
+  DatatypeSelectorDecl right("right", unresTree);
+  node.addSelector(right);
+  tree.addConstructor(node);
+
+  DatatypeConstructorDecl leaf("leaf");
+  DatatypeSelectorDecl data("data", unresList);
+  leaf.addSelector(data);
+  tree.addConstructor(leaf);
+
+  DatatypeDecl list = d_solver.mkDatatypeDecl("list");
+  DatatypeConstructorDecl cons("cons");
+  DatatypeSelectorDecl car("car", unresTree);
+  cons.addSelector(car);
+  DatatypeSelectorDecl cdr("cdr", unresTree);
+  cons.addSelector(cdr);
+  list.addConstructor(cons);
+
+  DatatypeConstructorDecl nil("nil");
+  list.addConstructor(nil);
+
+  std::vector<DatatypeDecl> dtdecls;
+  dtdecls.push_back(tree);
+  dtdecls.push_back(list);
+  std::vector<Sort> dtsorts;
+  TS_ASSERT_THROWS_NOTHING(dtsorts =
+                               d_solver.mkDatatypeSorts(dtdecls, unresTypes));
+  TS_ASSERT(dtsorts.size() == dtdecls.size());
+  for (unsigned i = 0, ndecl = dtdecls.size(); i < ndecl; i++)
+  {
+    TS_ASSERT(dtsorts[i].isDatatype());
+    TS_ASSERT(!dtsorts[i].getDatatype().isFinite());
+    TS_ASSERT(dtsorts[i].getDatatype().getName() == dtdecls[i].getName());
+  }
+  // verify the resolution was correct
+  Datatype dtTree = dtsorts[0].getDatatype();
+  DatatypeConstructor dtcTreeNode = dtTree[0];
+  TS_ASSERT(dtcTreeNode.getName() == "node");
+  DatatypeSelector dtsTreeNodeLeft = dtcTreeNode[0];
+  TS_ASSERT(dtsTreeNodeLeft.getName() == "left");
```

Comment:
Add `TS_ASSERT_THROWS` check for `getName()`.

Reply:
I can add a TS_ASSERT_THROWS_NOTHING (DatatypeDecl::getName can never throw anything, it should always have a non-null datatype by construction).

Reply:
Ah sorry, that was the wrong line. I meant to write this for this:https://github.com/CVC4/CVC4/pull/3930/files#diff-d4eff0da8e72b4f2d088f43ce4ecf899R1883